doc/configuration.md

```diff
@@ -0,0 +1,247 @@
+parent span ID, and the sampling decision.
+
+Different tracing systems have different standards for how trace context is
+propagated, e.g. which HTTP request headers are used.
+
+The Datadog C++ tracer supports two styles of trace context propagation.  The
+default style, `Datadog`, decodes trace information from multiple `X-Datadog-*`
+request headers.  For compatibility with [other tracing systems][7], another
+style, `B3`, is also supported.  The `B3` style decodes trace information from
+multiple `X-B3-*` request headers.
+
+The trace context extraction styles setting indicates which styles the tracer
+will consider when extracting trace context from a request.  At least one style
+must be specified, but multiple may be specified.  If multiple styles are
+specified, then trace context must be successfully extractable in at least one
+of the styles, and if trace context can be extracted in both styles, the two
+extracted contexts must agree.
+
+- **TracerOptions member**: `std::set<PropagationStyle> extract`
+- **JSON property**: `"propagation_style_extract"` _(array of string)_
+- **Environment variable**: `DD_PROPAGATION_STYLE_EXTRACT` _(JSON)_
+- **Default value**: `["Datadog"]`
+
+### Trace Context Injection Styles
+Trace context injection styles are analogous to trace context extraction styles
+(see the previous section), except that rather than indicating which trace
+context encoding are supported when _extracting_ trace context, trace context
+injection styles indicate which trace context encoding(s) will be used when
+_injecting_ context into a request to the next service along a trace.
+
+Note that even if the `B3` injection style is used, the tracer still may inject
+Datadog-specific trace context, such as in the `X-Datadog-Origin` request
+header.
+
+- **TracerOptions member**: `std::set<PropagationStyle> inject`
+- **JSON property**: `"propagation_style_inject"` _(array of string)_
+- **Environment variable**: `DD_PROPAGATION_STYLE_INJECT` _(JSON)_
+- **Default value**: `["Datadog"]`
+
+### Host Name Reporting
+If `true`, the tracer will look up its host's name on the network using the
+[gethostname][8] function and send it to the Datadog backend in a hidden span
+tag.
+
+- **TracerOptions member**: `bool report_hostname`
+- **JSON property**: `"dd.trace.report-hostname"` _(boolean)_
+- **Environment variable**: `DD_TRACE_REPORT_HOSTNAME`
+- **Default value**: `false`
+
+### Span Tags
+Tags to add to every span produced by the tracer.
+
+When specified as `std::map<std::string, std::string> tags`, each entry in the
+map is a (key, value) pair, where the key is the name of the span tag, and the
+value is its value.  The value is a string.
+
+When specified as the `DD_TAGS` environment variable, tags are formatted as a
+comma-separated list of `key:value` pairs (the key and value are separated by a
+colon).
+
+- **TracerOptions member**: `std::map<std::string, std::string> tags`
+- **JSON property**: `tags` _(object)_
```

Comment:
isn't this an array of something?

Reply:
It is a JSON object `{key: value}`.  The environment variable is sort of an array.

Reply:
https://github.com/DataDog/dd-opentracing-cpp/blob/80979a871759346bffa50f6fa215e3f87aed96b2/src/tracer_factory.cpp#L44-L46

https://github.com/DataDog/dd-opentracing-cpp/blob/80979a871759346bffa50f6fa215e3f87aed96b2/include/datadog/opentracing.h#L117